hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/StandardArgumentsValidators.java

```diff
@@ -26,79 +27,167 @@ private StandardArgumentsValidators() {
-				throw new QueryException(
-						String.format(
-								Locale.ROOT,
-								"Function requires %d arguments, but %d found",
-								number,
-								arguments == null ? 0 : arguments.size()
-						)
-				);
+
+			@Override
+			public String getSignature() {
+				StringBuilder sig = new StringBuilder("(");
+				for (int i=0; i<number; i++) {
+					if (i!=0) {
+						sig.append(", ");
+					}
+					sig.append("arg");
+					if (number>1) {
+						sig.append(i);
+					}
+				}
+				sig.append(")");
+				return sig.toString();
 			}
 		};
 	}
 
 	public static ArgumentsValidator max(int maxNumOfArgs) {
-		return arguments -> {
-			if ( arguments == null || arguments.size() > maxNumOfArgs ) {
-				throw new QueryException(
-						String.format(
-								Locale.ROOT,
-								"Function requires %d or fewer arguments, but %d found",
-								maxNumOfArgs,
-								arguments == null ? 0 : arguments.size()
-						)
-				);
+		return new ArgumentsValidator() {
+			@Override
+			public void validate(List<? extends SqmVisitableNode> arguments) {
+				if (arguments == null || arguments.size() > maxNumOfArgs) {
+					throw new QueryException(
+							String.format(
+									Locale.ROOT,
+									"Function requires %d or fewer arguments, but %d found",
+									maxNumOfArgs,
+									arguments == null ? 0 : arguments.size()
+							)
+					);
+				}
+			}
+
+			@Override
+			public String getSignature() {
+				StringBuilder sig = new StringBuilder("([");
+				for (int i=0; i<maxNumOfArgs; i++) {
+					if (i!=0) {
+						sig.append(", ");
+					}
+					sig.append("arg").append(i);
+				}
+				sig.append("])");
+				return sig.toString();
 			}
 		};
 	}
 
 	public static ArgumentsValidator between(int minNumOfArgs, int maxNumOfArgs) {
-		return arguments -> {
-			if ( arguments == null || arguments.size() < minNumOfArgs || arguments.size() > maxNumOfArgs ) {
-				throw new QueryException(
-						String.format(
-								Locale.ROOT,
-								"Function requires between %d and %d arguments, but %d found",
-								minNumOfArgs,
-								maxNumOfArgs,
-								arguments == null ? 0 : arguments.size()
-						)
-				);
+		return new ArgumentsValidator() {
+			@Override
+			public void validate(List<? extends SqmVisitableNode> arguments) {
+				if (arguments == null || arguments.size() < minNumOfArgs || arguments.size() > maxNumOfArgs) {
+					throw new QueryException(
+							String.format(
+									Locale.ROOT,
+									"Function requires between %d and %d arguments, but %d found",
+									minNumOfArgs,
+									maxNumOfArgs,
+									arguments == null ? 0 : arguments.size()
+							)
+					);
+				}
+			}
+			
+			@Override
+			public String getSignature() {
+				StringBuilder sig = new StringBuilder("(");
+				for (int i=0; i<maxNumOfArgs; i++) {
+					if (i==minNumOfArgs) {
+						sig.append("[");
+					}
+					if (i!=0) {
+						sig.append(", ");
+					}
+					sig.append("arg").append(i);
+				}
+				sig.append("])");
+				return sig.toString();
```

Comment:
I am not sure whether my comment makes sense, but it seems the space delimiter code style is not followed strictly in many places in this PR. Maybe inevitable when tons of new code is created.